lib/ledger_sync/adaptors/quickbooks_online/vendor/operations/create.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+module LedgerSync
+  module Adaptors
+    module QuickBooksOnline
+      module Vendor
+        module Operations
+          class Create < Operation::Create
+            class Contract < LedgerSync::Adaptors::Contract
+              params do
+                required(:ledger_id).value(:nil)
+                optional(:first_name).maybe(:string)
+                optional(:last_name).maybe(:string)
+                optional(:email).maybe(:string)
+              end
+            end
+
+            private
+
+            def operate
+              response = adaptor.upsert(
+                resource: 'vendor',
+                payload: local_resource_data
+              )
+
+              resource.ledger_id = response.dig('Id')
+              success(response: response)
+            end
+
+            def local_resource_data
+              {
+                'GivenName' => resource.first_name,
+                'FamilyName' => resource.last_name,
+                'PrimaryEmailAddr' => {
+                  'Address' => resource.email
+                }
```

Comment:
Should we only attach the email stuff if `resource.email != nil`? Or does it not matter?

Reply:
I would do this in different PR. Its about knowing what attributes are dirty. Like right now we fetch latest value from QBO and then override with our payload. Which allows us to set empty values.

But if we don't provide value, does it mean we wanna use its latest value or we wanna clear up the value? Which is why we need to know on resource if attribute is dirty or no. If its empty because we set it to be empty or because we didnt provide value.

Reply:
Tested it earlier on Customer above. Not sending anything is same as sending empty value. QBO removes it.

Reply:
Yes, this is tracked in #8 .  Probably becoming a priority to define this behavior soon.